### PR TITLE
Fix some resolution planning issues

### PR DESCRIPTION
**Pull request by kasper-piskorski**
This PR solves an issue of omittance of value predicates when doing planning. Additionally it fixes a problem with unifying subbed atoms with themselves. However, this PR will make some of the queries slower due to the fact that it groups relation with same type together. This is fixed here: 

https://github.com/graknlabs/grakn/pull/2382

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -108,6 +108,41 @@ def buildGrakn() {
...
+      `>>%%%%>>> `:::`%%%%%%%%%%>;.     ;;>>%%%%%                   ;>%%'
+       `>>%%%%>>, `::::`%%%%%%%%%%>,   .;>>%%%%%'                   ;>%'
+        `>>%%%%>>, `:::::`%%%%%%%%%>>. ;;>%%%%%%                    ;>%,
+         `>>%%%%>>, :::::::`>>>%%%%>>> ;;>%%%%%'                     ;>%,
+          `>>%%%%>>,::::::,>>>>>>>>>>' ;;>%%%%%                       ;%%,
+            >>%%%%>>,:::,%%>>>>>>>>'   ;>%%%%%.                        ;%%
+             >>%%%%>>``%%%%%>>>>>'     `>%%%%%%.
+             >>%%%%>> `@@a%%%%%%'     .%%%%%%%%%.
+             `a@@a%@'    `%a@@'       `a@@a%a@@a'
+     */
```
Comment (aelred) [inline Jenkinsfile]: I need to remove this before we merge

File: grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/AtomicBase.java [hunk 2]
```diff
@@ -82,8 +83,13 @@ protected AtomicBase(AtomicBase a) {
     @Override
     public VarPattern getPattern(){ return atomPattern;}
 
+    protected Pattern combinedPattern(){ return atomPattern;}
```
Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/AtomicBase.java]: Maybe this should be `createCombinedPattern()` or something? We need to distinguish it from `getCombinedPattern()` below.

File: grakn-graql/src/test/java/ai/grakn/graql/internal/hal/HALBuilderTest.java [hunk 3]
```diff
@@ -115,7 +115,7 @@ public void whenExplainInferred_returnsLinkedExplanation() {
             String specificQuery = "match " +
                     "$x id " + cousin1 + ";" +
                     "$y id " + cousin2 + ";" +
-                    "(cousin: $x, cousin: $y) isa cousins; get;";
+                    "(cousin: $x, cousin: $y) isa cousins; limit 1;get;";
```
Comment (aelred) [inline grakn-graql/src/test/java/ai/grakn/graql/internal/hal/HALBuilderTest.java]: ahhhh... that's why this was slow :P

**Comment by kasper-piskorski** [comment]
@aelred ready to merge?

**Pull request merged**